charts/otomi-api/templates/deployment.yaml

```diff
@@ -19,6 +19,11 @@ spec:
         checksum/secret: {{ include (print .Template.BasePath "/secret.yaml") . | sha256sum | trunc 63 }}
         checksum/config: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum | trunc 63 }}
         checksum/core: {{ include (print .Template.BasePath "/core-config.yaml") . | sha256sum | trunc 63 }}
+        {{- with .Values.podAnnotations }}
```

Comment:
no need to use with as you are already in annotations block...please understand what you are doing here.

Just iterate over the range, and it will print nothing when it is empty